### DISCUSSION TITLE
$http.send ignore SSL error for self-signed ssl

### DISCUSSION THREAD

**Discussion by copongcopong** [pocketbase/pocketbase#5968] [Q&A] [answered]
Is there a config in $http.send to ignore SSL certificate errors? I am trying to reach url with self-signed certificate.

**Comment by ganigeorgiev** [comment, marked as answer]
No (see also https://github.com/pocketbase/pocketbase/discussions/5553).

Instead of trying to workaround it, my recommendation would be to load the certificate on the system where PocketBase is running, otherwise it kind of defeats the purpose of having a certificate in the first place.

**Comment by copongcopong** [comment]
For go/pocketbase `$http` to recognize a domain that has a self-signed certificate, provided that you also have access to it, 
the self-signed certificated *must* be generated using SANs. 

in the steps below, make sure to replace `ec2-domain.domain.com` with the proper domain name

Step 1: self-signed certificate generation 

```
openssl genrsa -out private.key 2048 

openssl req -new -x509 -nodes -sha1 -days 365 -extensions v3_ca -key private.key -out ec2.crt -subj '/CN=ec2-domain.domain.com'   -addext 'subjectAltName=DNS:ec2-domain.domain.com'
```
Then use this in your https config (e.g. nginx conf file)

Step 2:
Setup the machine (ubuntu) to trust the `.crt`; copy paste the contents of the `.crt` using the command below

```
sudo nano /usr/local/share/ca-certificates/the-self-signed-certificate-from-other-api-or-server.crt 

sudo chmod 0600 /usr/local/share/ca-certificates/the-self-signed-certificate-from-other-api-or-server.crt

sudo cp /usr/local/share/ca-certificates/the-self-signed-certificate-from-other-api-or-server.crt /usr/share/ca-certificates/the-self-signed-certificate-from-other-api-or-server.crt

echo 'signed-certificate-from-other-api-or-server.crt' | sudo tee -a /etc/ca-certificates.conf

sudo update-ca-certificates 
```

`$http.send` will have no errors

you can also test this via curl like `curl -X GET https://ec2-domain.domain.com/endpoint/test`  and should have a response without errors
